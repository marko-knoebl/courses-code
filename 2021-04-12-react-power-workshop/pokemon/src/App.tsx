/** @jsxImportSource @emotion/react */
import { css } from "@emotion/react";
import { NavLink, Routes, Route } from "react-router-dom";

import "./styles.css";
import HomeView from "./home/HomeView";
import PokemonView from "./pokemon/PokemonView";
import CartView from "./cart/CartView";
import { useCart } from "./useCart";
import { usePokemons } from "./usePokemons";
import CartContext from "./CartContext";
import DocumentTitle from "./components/DocumentTitle";
import CounterThatLogsEverySecond from "./CounterThatLogsEverySecond";

function App() {
  const { cart, addToCart, removeFromCart } = useCart();

  const { loadedPokemons, loadMore, requestPokemon } = usePokemons();

  return (
    <CartContext.Provider
      value={{
        cart: cart,
        addToCart: addToCart,
        removeFromCart: removeFromCart,
      }}
    >
      <DocumentTitle
        value={`Pokémon Poster Shop (${cart.length} items in the cart)`}
      />
      <CounterThatLogsEverySecond />
      <div
        css={css`
          background-color: #f0f0f0;
        `}
      >
        <h1>Pokémon poster shop</h1>
        <NavLink to="/">Home</NavLink> |{" "}
        <NavLink to="/cart">
          {cart.length === 0 ? "Cart" : `Cart (${cart.length})`}
        </NavLink>
        <Routes>
          <Route
            path="/"
            element={<HomeView pokemons={loadedPokemons} loadMore={loadMore} />}
          />
          <Route
            path="/pokemon/:id"
            element={
              <PokemonView
                pokemons={loadedPokemons}
                requestPokemon={requestPokemon}
                addToCart={addToCart}
                onRemoveFromCart={removeFromCart}
                cart={cart}
              />
            }
          />
          <Route
            path="/cart"
            element={<CartView pokemons={loadedPokemons} />}
          />
        </Routes>
      </div>
    </CartContext.Provider>
  );
}

export default App;
