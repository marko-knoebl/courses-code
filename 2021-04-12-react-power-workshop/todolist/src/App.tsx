import { useEffect, useReducer, useState } from "react";
import AddTodo from "./AddTodo";
import AddTodoClass from "./AddTodoClass";
import Rating from "./Rating";
import { Todo } from "./Todo";
import TodoItem from "./TodoItem";
import { fetchTodos } from "./todosApi";
import { NavLink, Routes, Route } from "react-router-dom";
import AboutPage from "./AboutPage";

/*
Wir erstellen eine Todo-Anwendung mit der folgenden Funktionalität:

+ Anzeigen erledigter und nicht-erledigter Todos
+ Hinzufügen eines Todos mittels eines Formulars
+ Umschalten des erledigt-Zustandes eines Todos
- Löschen eines Todos
*/

const initialTodos: Array<Todo> = [
  { id: 1, title: "Einkaufen", completed: false },
  { id: 2, title: "Steuererklärung", completed: true },
  { id: 3, title: "Aufräumen", completed: false },
];

type AddTodoAction = {
  type: "addTodo";
  payload: string;
};
type ChangeCompletedAction = {
  type: "changeCompleted";
  payload: { id: number; completed: boolean };
};
type DeleteTodoAction = {
  type: "deleteTodo";
  payload: number;
};
type LoadTodosAction = {
  type: "loadTodos";
  payload: Array<Todo>;
};
type TodosAction =
  | AddTodoAction
  | ChangeCompletedAction
  | DeleteTodoAction
  | LoadTodosAction;

function todosReducer(
  todosState: Array<Todo>,
  action: TodosAction
): Array<Todo> {
  switch (action.type) {
    case "addTodo":
      const newId = Math.max(...todosState.map((t) => t.id), 0) + 1;
      return [
        ...todosState,
        { id: newId, title: action.payload, completed: false },
      ];
    case "deleteTodo":
      return todosState.filter((todo) => todo.id !== action.payload);
    case "changeCompleted":
      return todosState.map((todo) =>
        todo.id === action.payload.id
          ? { ...todo, completed: action.payload.completed }
          : todo
      );
    case "loadTodos":
      return action.payload;
    default:
      return todosState;
  }
}

function App() {
  const [rating1, setRating1] = useState(3);

  const [todos, dispatch] = useReducer(todosReducer, initialTodos);

  function addTodo(title: string): void {
    dispatch({ type: "addTodo", payload: title });
  }

  function changeCompleted(id: number, completed: boolean): void {
    dispatch({
      type: "changeCompleted",
      payload: { id: id, completed: completed },
    });
  }

  function deleteTodo(id: number): void {
    dispatch({ type: "deleteTodo", payload: id });
  }

  async function loadTodosFromAPI() {
    const apiTodos = await fetchTodos();
    dispatch({ type: "loadTodos", payload: apiTodos });
  }
  // useEffect(() => {
  //   loadTodosFromAPI();
  // }, []);

  function saveTodosToLS() {
    localStorage.setItem("todos", JSON.stringify(todos));
  }
  function loadTodosFromLS() {
    const lsTodosString = localStorage.getItem("todos");
    if (lsTodosString) {
      dispatch({ type: "loadTodos", payload: JSON.parse(lsTodosString) });
    }
  }
  useEffect(() => {
    saveTodosToLS();
  }, [todos]);

  return (
    <div>
      <Rating value={rating1} onChange={setRating1} />
      <h1>Todo</h1>
      <NavLink to="/">home</NavLink> <NavLink to="/about">about</NavLink>
      <Routes>
        <Route
          path="/"
          element={
            <div>
              <div>
                <button onClick={loadTodosFromAPI}>load todos from API</button>
                <button onClick={saveTodosToLS}>
                  save todos to local storage
                </button>
                <button onClick={loadTodosFromLS}>
                  load todos from local storage
                </button>
              </div>
              <ul style={{ maxHeight: 400, overflow: "auto" }}>
                {todos.map((todo) => (
                  <TodoItem
                    key={todo.id}
                    title={todo.title}
                    completed={todo.completed}
                    onCompletedChange={(completed) => {
                      changeCompleted(todo.id, completed);
                    }}
                    onDelete={() => deleteTodo(todo.id)}
                  />
                ))}
              </ul>
              <AddTodo onAdd={(newTitle) => addTodo(newTitle)} />
              <AddTodoClass onAdd={(newTitle) => addTodo(newTitle)} />
            </div>
          }
        />
        <Route path="/about" element={<AboutPage />} />
      </Routes>
    </div>
  );
}

export default App;
