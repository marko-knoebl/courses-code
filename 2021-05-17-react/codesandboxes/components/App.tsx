import "./styles.css";
import { useState } from "react";
import ProgressBar from "./ProgressBar";
import Rating from "./Rating";

export default function App() {
  const [prog, setProg] = useState(0.75);
  const [prog2, setProg2] = useState(0.5);
  const [rating, setRating] = useState(3);
  const [rating2, setRating2] = useState(4);

  return (
    <div className="App">
      <h1>Progress bar</h1>
      <ProgressBar value={prog} />
      <ProgressBar value={prog2} color="green" />
      <h1>Rating</h1>
      <Rating value={rating} onChange={(newRating) => setRating(newRating)} />
      <Rating
        value={rating2}
        onChange={(newRating2) => setRating2(newRating2)}
      />
      <div>average rating: {(rating + rating2) / 2}</div>
    </div>
  );
}
