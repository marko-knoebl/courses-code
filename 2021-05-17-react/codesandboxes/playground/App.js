import React, { useState } from "react";
import { FontSizeDemo } from "./FontSizeDemo";

const baseUrl = "https://picsum.photos/300/200?image=";
function SlideshowApp() {
  // Slideshow

  const [img, setImg] = useState(0);

  const imgUrl = `${baseUrl}${img}`;

  const buttonStyle = {
    color: "white",
    backgroundColor: "navy"
  };

  function prevImg() {
    if (img > 0) {
      setImg(img - 1);
    }
  }

  function randImg() {
    setImg(Math.floor(Math.random() * 100));
  }

  // Passwords

  const [password1, setPassword1] = useState("");
  const [password2, setPassword2] = useState("");
  let message;
  if (!(password1.length >= 4)) {
    message = "Pasword 1 is not long enough.";
  } else if (!(password1 === password2)) {
    message = "Passwords don't match.";
  } else {
    message = "";
  }

  // react methods

  const reactMethods = [];
  for (let method in React) {
    reactMethods.push(<li>{method}</li>);
  }

  // todo list

  const initialTodos = [
    { id: 1, title: "groceries", completed: false },
    { id: 2, title: "cooking", completed: true },
    { id: 3, title: "gardening", completed: false }
  ];
  const [todos, setTodos] = useState(initialTodos);

  return (
    <>
      <FontSizeDemo />
      <h1>Image {img}</h1>
      <div style={{ backgroundColor: "lightgrey", display: "flex" }}>
        <button style={buttonStyle} onClick={() => setImg(0)}>
          start
        </button>
        <button style={buttonStyle} onClick={prevImg}>
          prev
        </button>
        <img src={imgUrl} alt="slideshow" style={{ display: "block" }} />
        <button style={buttonStyle} onClick={() => setImg(img + 1)}>
          next
        </button>
        <button style={buttonStyle} onClick={randImg}>
          random
        </button>
      </div>
      <div>
        <label>
          password:
          <input
            value={password1}
            onChange={(event) => setPassword1(event.target.value)}
          />
        </label>
        <label>
          confirm password:
          <input
            value={password2}
            onChange={(event) => setPassword2(event.target.value)}
          />
        </label>
        <div>{message}</div>
        {password1 === password2 ? "Passwords match" : "passwords don't match"}
      </div>
      <h2>React props</h2>
      <ul>{reactMethods}</ul>
      <h2>Todos</h2>
      <ul>
        {todos.map((todo) => (
          <li key={todo.id}>{todo.title}</li>
        ))}
      </ul>
    </>
  );
}
export default SlideshowApp;

// Datenverwaltung ohne Mutationen

// Einzelnes Todo

function withCompletedToggled(todo) {
  return { ...todo, completed: !todo.completed };
}

function withTitleChanged(todo, newTitle) {
  return { ...todo, title: newTitle };
}

const todo1 = { id: 1, title: "foo", completed: false };

const todo2 = withCompletedToggled(todo1);
// { id: 1, title: 'foo', completed: true }

const todo3 = withTitleChanged(todo2, "bar");
// { id: 1, title: 'bar', completed: true}

console.log(todo1);
console.log(todo2);
console.log(todo3);

// Array von Todos

function withNewTodoAdded(todos, newTitle) {
  let maxId = 0;
  for (let todo of todos) {
    if (todo.id > maxId) {
      maxId = todo.id;
    }
  }

  return [...todos, { id: maxId + 1, title: newTitle, completed: false }];
}

function withTodoToggled(todos, id) {
  return todos.map((todo) => {
    if (todo.id === id) {
      // "abändern"
      return { ...todo, completed: !todo.completed };
    } else {
      // nicht "abändern"
      return todo;
    }
  });
}

function withCompletedTodosRemoved(todos) {
  return todos.filter((todo) => !todo.completed);
}

const todos1 = [
  { id: 1, title: "foo", completed: false },
  { id: 2, title: "bar", completed: true }
];

const todos2 = withNewTodoAdded(todos1, "baz");
console.log(todos2);

// schaltet den "completed" - state des todos mit der id 1 um
const todos3 = withTodoToggled(todos2, 1);
console.log(todos3);

const todos4 = withCompletedTodosRemoved(todos3);
console.log(todos4);
