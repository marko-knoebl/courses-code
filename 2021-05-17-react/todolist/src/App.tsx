import { useState } from "react";
import { AppBar, Button } from "@material-ui/core";
import { Theme } from "./types";
import AddTodo from "./AddTodoCls";
import TodoList from "./TodoList";
import AboutView from "./AboutView";
import { NavLink, Route, Routes } from "react-router-dom";
import ThemeContext from "./ThemeContext";
import { useTodos } from "./useTodos";

function App() {

  const {
    todos,
    isLoading,
    addTodo,
    toggleTodo,
    deleteTodo,
    loadTodosAsync,
  } = useTodos()

  // theme

  const [theme, setTheme] = useState<Theme>("light");

  return (
    <ThemeContext.Provider value={theme}>
      <div style={{ backgroundColor: theme === "light" ? "#eee" : "#666" }}>
        <AppBar position="static">
          <h1>Todo</h1>
        </AppBar>

        <nav>
          <NavLink to="/">Home</NavLink> <NavLink to="/about">About</NavLink>
        </nav>

        <Routes>
          <Route
            path="/"
            element={
              <>
                <div>
                  <Button variant="contained" onClick={loadTodosAsync}>
                    load from API
                  </Button>
                  {isLoading ? <div>Loading...</div> : null}
                </div>
                <TodoList
                  todos={todos}
                  onToggle={toggleTodo}
                  onDelete={deleteTodo}
                />
                <AddTodo onAddTodo={addTodo} />
              </>
            }
          />
          <Route path="/about" element={<AboutView />} />
        </Routes>
      </div>
    </ThemeContext.Provider>
  );
}

export default App;
