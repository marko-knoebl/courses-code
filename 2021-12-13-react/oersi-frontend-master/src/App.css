@import "../node_modules/bootstrap/dist/css/bootstrap.min.css";

* {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Noto Sans",
    "Ubuntu", "Droid Sans", "Helvetica Neue", sans-serif;
}

:root {
  font-size: 13px !important;
  line-height: 24px !important;
  vertical-align: baseline !important;
  font-weight: 400 !important;
  font-style: normal !important;
  font-variant: normal !important;
  text-decoration: none solid rgba(24, 28, 43) !important;
}

body {
  background-color: #c1c2c3 !important;
}

.reactive-base {
  font-size: 13px !important;
}

.search-component-input {
  border-radius: 1em;
  background-color: #f7f7f7;
  margin-bottom: 0px;
  border: 1px solid #ced4da;
  font-size: 1rem !important;
}

.search-component-input:focus {
  color: #495057;
  background-color: #f7f7f7 !important;
  border-color: #f8d0d0;
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(232, 97, 97, 0.25);
}

.full-width {
  width: 100%;
  width: -webkit-fill-available; /* Mozilla-based browsers will ignore this. */
  width: fill-available;
  width: -moz-available; /* WebKit-based browsers will ignore this. */
}
