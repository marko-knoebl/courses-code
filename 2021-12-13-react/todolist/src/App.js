import { useState } from "react";

const initialTodos = [
  { id: 1, title: "groceries", completed: false },
  { id: 2, title: "cooking", completed: true },
  { id: 3, title: "gardening", completed: false },
];

const App = () => {
  const [todos, setTodos] = useState(initialTodos);
  return (
    <ul>
      {todos.map((todo) => (
        <li
          key={todo.id}
          style={{ backgroundColor: todo.completed ? "lightgrey" : "salmon" }}
        >
          {(todo.completed ? "done" : "todo") + ": "}
          {todo.title}
        </li>
      ))}
    </ul>
  );
};

export default App;
