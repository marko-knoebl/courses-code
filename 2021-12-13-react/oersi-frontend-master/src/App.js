import React, {useState, useEffect} from "react"
import {ReactiveBase} from "@appbaseio/reactivesearch"
import {BrowserRouter as Router, Route, Switch} from "react-router-dom"
import "./App.css"
import Cookie from "./components/cookieComponent/Cookie"
import Contact from "./components/contact/Contact"
import FooterComponent from "./components/footerComponent/FooterComponent"
import HeaderComponent from "./components/headerComponent/HeaderComponent"
import ResourceDetails from "./components/resourceDetails/ResourceDetails"
import SearchIndexView from "./components/SearchIndexView"
import {ScrollTop} from "./helpers/ScrollTop"
import {ConfigurationRunTime} from "./helpers/use-context"

const App = (props) => {
  const context = React.useContext(ConfigurationRunTime)
  const [multilist] = useState(props.config.get("multiList"))
  // breakpoints - see https://getbootstrap.com/docs/4.0/layout/overview/#responsive-breakpoints
  const isMobileOrTablet = useMedia("(max-width: 991.98px)")

  return (
    <Router basename={process.env.PUBLIC_URL}>
      <ReactiveBase
        className="reactive-base"
        app={props.elasticSearch.APP_NAME}
        url={props.elasticSearch.URL}
        headers={getAuthorizationHeaderIfCredentialsExist(
          props.elasticSearch.CREDENTIALS
        )}
      >
        <HeaderComponent />
        {context.FEATURES.SCROLL_TOP_BUTTON && <ScrollTop />}
        <Switch>
          <Route exact path="/">
            <SearchIndexView isMobile={isMobileOrTablet} multilist={multilist} />
          </Route>
          <Route exact path="/services/contact" component={Contact} />
          <Route
            exact
            path="(/details)?/:resourceId([A-Za-z0-9-_=]{12,})"
            component={ResourceDetails}
          />
        </Switch>
        <FooterComponent />
        <Cookie />
      </ReactiveBase>
    </Router>
  )

  /**
   * function that returns the authorization-header if credentials exist
   * @param {String} credentials
   */
  function getAuthorizationHeaderIfCredentialsExist(credentials) {
    if (credentials !== "" && credentials) return {authorization: credentials}
    else return null
  }

  function useMedia(query) {
    const [matches, setMatches] = useState(false)

    // Activity normally for componentDidMount + componentDidUpdate
    useEffect(() => {
      const media = window.matchMedia(query)
      if (media.matches !== matches) {
        setMatches(media.matches)
      }

      const listener = () => setMatches(media.matches)
      media.addListener(listener)

      return () => media.removeListener(listener)
    }, [query, matches])

    // publish value for render
    return matches
  }
}

export default App
