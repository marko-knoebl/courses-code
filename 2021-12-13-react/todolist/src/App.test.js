import { render, screen } from "@testing-library/react";
import App from "./App";

it("matches the snapshot", () => {
  render(<App />);
  const ul = screen.getByRole("list");
  expect(ul).toMatchSnapshot();
});
