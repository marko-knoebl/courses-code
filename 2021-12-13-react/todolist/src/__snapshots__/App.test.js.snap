// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<ul>
  <li
    style="background-color: salmon;"
  >
    todo: 
    groceries
  </li>
  <li
    style="background-color: lightgrey;"
  >
    done: 
    cooking
  </li>
  <li
    style="background-color: salmon;"
  >
    todo: 
    gardening
  </li>
</ul>
`;
